frontend/widgets/OBSBasic_SceneItems.cpp

```diff
@@ -369,7 +369,21 @@ void OBSBasic::SetDeinterlacingMode()
 	OBSSceneItem sceneItem = GetCurrentSceneItem();
 	obs_source_t *source = obs_sceneitem_get_source(sceneItem);
 
+	obs_deinterlace_mode oldMode = obs_source_get_deinterlace_mode(source);
 	obs_source_set_deinterlace_mode(source, mode);
+
+	auto undo_redo = [](const std::string &uuid, obs_deinterlace_mode val) {
+		OBSSourceAutoRelease source = obs_get_source_by_uuid(uuid.c_str());
+		if (source)
+			obs_source_set_deinterlace_mode(source, val);
+	};
+
+	const char *uuid = obs_source_get_uuid(source);
```

Comment:
The `obs_source_get_uuid` API call can return a `NULL` pointer, and `undo_stack::add_action` takes const `std::string` references.

The compiler will implicitly create two temporary `std::string` instances initialised with this `const char` pointer, but that will crash with a `nullptr`.

If this is desired (the program is supposed to just crash in that situation) the code is correct, but if this is not a place where it should "just crash", a `nullptr` check is necessary here.

And because there are no locks around the source object as far as I can tell and other threads can potentially interfere with the shared `obs` instance, this value (and the source object state) can change, even between function calls.